crates/test-framework/src/spicetest/consistency/mod.rs

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::{
+    sync::Arc,
+    time::{Duration, SystemTime},
+};
+
+use async_openai::{
+    config::OpenAIConfig,
+    types::{
+        ChatCompletionRequestMessage, ChatCompletionRequestUserMessageArgs,
+        CreateChatCompletionRequest, CreateChatCompletionRequestArgs, CreateEmbeddingRequest,
+        EncodingFormat,
+    },
+    Client as OpenAIClient,
+};
+use reqwest::Client;
+use worker::{ConsistencyWorker, ConsistencyWorkerResult, WorkerHandle};
+
+use crate::{
+    metrics::{NoExtendedMetrics, QueryMetric},
+    spiced::SpicedInstance,
+};
+
+mod worker;
+
+#[derive(Clone)]
+pub struct ConsistencyConfig {
+    /// The total duration of the test.
+    pub duration: Duration,
+
+    /// The number of buckets to divide the test duration into.
+    pub buckets: usize,
+
+    /// The number of individial HTTP clients to make requests in parallel.
+    pub concurrency: usize,
+
+    /// The payloads to send to the component, specifically to be used in [`HttpConsistencyComponent::send_request`].
+    pub payloads: Vec<Arc<str>>,
+
+    /// The HTTP component, within the Spiced instance, to test.
+    pub component: HttpConsistencyComponent,
+}
+
+/// A component within the Spiced instance to test for consistency.
+///
+/// This component must be accessible over HTTP.
+#[derive(Clone)]
+pub enum HttpConsistencyComponent {
```

Comment:
Might be cleaner to make this a trait with separate implementations? Otherwise it might get unwieldy with more than a couple of variants